### PR TITLE
Implement expand_dims

### PR DESCRIPTION
**Pull request by okuta**
It is related to #337.

### CODE REVIEW

File: cupy/manipulation/dims.py [hunk 1]
```diff
@@ -183,8 +183,24 @@ def broadcast_arrays(*args):
...
+        cupy.ndarray: The number of dimensions is one greater than that of
+            the input array.
+
+    .. seealso:: :func:`numpy.expand_dims`
+
+    """
+    a = cupy.asarray(a)
+    shape = a.shape
+    if axis < 0:
+        axis = axis + len(shape) + 1
```
Comment (delta2323) [inline cupy/manipulation/dims.py]: What happens when `axis + len(shape) + 1` is still negative?

**Comment by okuta** [comment]
I added negative test case.

File: cupy/manipulation/dims.py (hunk 1 above)
Comment (beam2d) [inline cupy/manipulation/dims.py]: It should delegate the handling of negative axes to tuple indexing instead of handling by hand.

**Comment by okuta** [comment]
> What happens when axis + len(shape) + 1 is still negative?

Slice can specify axis of the out of range.

And, this implementation has numpy compatible.
https://github.com/numpy/numpy/blob/v1.9.1/numpy/lib/shape_base.py#L273

**Comment by okuta** [comment]
> What happens when axis + len(shape) + 1 is still negative?

Even if the axis is out of range , Slice will work.

And, this implementation has numpy compatible.
https://github.com/numpy/numpy/blob/v1.9.1/numpy/lib/shape_base.py#L273

**Comment by delta2323** [comment]
@okuta Thank you for adding negative indices cases.

**Comment by beam2d** [comment]
OK, I'll merge.

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**